client/my-sites/checkout/concierge-quickstart-session/index.jsx

```diff
@@ -45,10 +45,19 @@ import './style.scss';
 
 export class ConciergeQuickstartSession extends React.Component {
 	static propTypes = {
-		receiptId: PropTypes.number,
-		selectedSiteId: PropTypes.number.isRequired,
+		receiptId: PropTypes.string,
+		selectedSiteId: PropTypes.number,
 	};
 
+	componentDidMount() {
+		// If the param value is 'add', then we will add the concierge session item to cart and proceed to checkout,
+		// without rendering the UI. We use this method when a logged out user clicks 'Get Started' on the concierge
+		// offer page.
+		if ( 'add' === this.props.receiptId ) {
+			this.handleClickAccept();
+		}
```

Comment:
If you don't want to render any UI at all for the `../add` route, you can add the product to cart and redirect to checkout in the page.js handler (`conciergeQuickstartSession` function in `my-sites/checkout/controller.jsx`). And render the `ConciergeQuickstartSession` component only if there's a numeric `receiptId` and an actual UI to render.

Reply:
Good point. In [9849222](https://github.com/Automattic/wp-calypso/pull/33749/commits/98492222ecd61d3624a0558dcc8394047785e873), I've handled this in a slightly different (and hopefully better) way by removing the `/add` route altogether. Instead, a redirect happens to `/checkout/SITE_SLUG/quickstart-session`, so that the controller for the /checkout route( `function checkout ()` in `my-sites/checkout/controller.jsx`) will call the `Checkout` component which can handle addition of new products, similar to routes such as `/checkout/SITE_SLUG/business` and `/checkout/SITE_SLUG/premium`. 

One slight difference introduced by this method is that after clicking `Yes, I want a WordPress expert` button OR `Get Started` button(if logged out) in the [offer page](https://user-images.githubusercontent.com/1269602/59348913-000b4600-8d36-11e9-9942-a639660274c9.png), it will redirect to `/checkout/SITE_SLUG/quickstart-session` instead of the earlier `/checkout/SITE_SLUG`. I'm not sure if this is a problem, but just making a note here.